### PR TITLE
Implement all intrinsic function

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
9116df3

### CODE REVIEW

File: integration_tests/intrinsics_47.f90 [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    logical :: a(3) = [.true., .true., .true.]
+    l = all([.true., .false.])
+    l = all([l])
+    l = all([.false.])
+    l = all([.true.])
+    l = all([1 == 2, 0 == 0])
+    l = all(a)
+    l = all([.false., a, .true.])
+    if (l) error stop
+end program intrinsics_47
```
Comment (czgdp1807) [inline integration_tests/intrinsics_47.f90:13]: Please register it with `gfortran` (`llvm` too if its working) in `integration_tests/CMakeLists.txt`.

File: tests/tests.toml [hunk 2]
```diff
@@ -1745,6 +1745,10 @@ asr = true
 filename = "../integration_tests/intrinsics_46.f90"
 asr = true
 
+[[test]]
+filename = "../integration_tests/intrinsics_47.f90"
+asr = true
```
Comment (czgdp1807) [inline tests/tests.toml:1750]: Remove if working with `llvm`.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

**Review by certik — APPROVED** [review]
I think this is good to merge as is. We can improve upon this later.

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 3]
```diff
@@ -1171,8 +1171,17 @@ LFORTRAN_API int32_t _lfortran_ichar(char *c) {
...
+}
+
+LFORTRAN_API int32_t _lfortran_all(bool *mask, int32_t n) {
+    for (size_t i = 0; i < n; i++) {
+        if (!mask[i]) {
+            return false;
+        }
+    }
+    return true;
+}
```
[the remark this answers is not in the document]
Comment (certik) [reply src/libasr/runtime/lfortran_intrinsics.c:1184]: We'll have to figure out and document some consistent way to handle these runtime functions, there will be a lot of them. It seems doing these in the surface language and use them via an ASR pass makes a lot of sense, see the IntrinsicFunction work. For now calling from C I think is fine.

Direct LLVM implementation is good, but less readable, and it should be equivalent to the surface level implementation (later on). Obviously the C implementation is slower, but it's the easiest for us to maintain for now, so I think this is ok as an intermediate step.

File: tests/tests.toml (hunk 2 above)
Comment (certik) [reply tests/tests.toml:1759]: It works with LLVM, but the integration test does not test the validity of the output, so we still need this until the test is improved.

File: src/libasr/ASR.asdl [hunk 4]
```diff
@@ -313,6 +313,7 @@ expr
     | IntegerBitLen(expr a, ttype type, expr? value)
     | Ichar(expr arg, ttype type, expr? value)
     | Iachar(expr arg, ttype type, expr? value)
+    | All(expr arg, expr? dim, ttype type, expr? value)
```
Comment (certik) [inline src/libasr/ASR.asdl:316]: This should be called `ArrayAll`, shouldn't it?

Comment (Thirumalai-Shaktivel) [reply src/libasr/ASR.asdl:316]: Yup, I think so, as `all` accepts logical non-scalar `mask` argument

**Pull request merged**